### PR TITLE
Add UserAssigned Identity to aks module

### PR DESCRIPTION
**Pull request by p3ck**
##### SUMMARY
Adds the ability to assign User Assigned Identity to the Azure Kubernetes Service.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
azure_rm_aks

##### ADDITIONAL INFORMATION
Currently the aks module only supports a service_principal or System Assigned identity.  This adds the ability to assign a User Assigned Identity to the cluster.

Integration tests have been updated to validate this feature.

### CODE REVIEW

File: plugins/modules/azure_rm_aks.py [hunk 1]
```diff
@@ -590,6 +611,9 @@
         provisioning_state: Succeeded
         service_principal_profile:
            client_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+        identity:
+           "type": "UserAssigned",
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_aks.py]:
```suggestion
           "type": "UserAssigned"
```

File: plugins/modules/azure_rm_aks.py [hunk 2]
```diff
@@ -938,6 +985,8 @@ def __init__(self):
         self.node_resource_group = None
         self.pod_identity_profile = None
 
+        mutually_exclusive = [('identity','service_principal')]
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_aks.py]: missing whitespace after ','
```suggestion
        mutually_exclusive = [('identity', 'service_principal')]
```

File: plugins/modules/azure_rm_aks.py [hunk 3]
```diff
@@ -1387,6 +1450,35 @@ def create_addon_profile_instance(self, addon):
         return result
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_aks.py]: too many blank!
```suggestion
```

File: plugins/modules/azure_rm_aks.py [hunk 4]
```diff
@@ -1387,6 +1450,35 @@ def create_addon_profile_instance(self, addon):
...
+        # If type set to SystamAssigned, and Resource has SystamAssigned, nothing to do
+        if 'SystemAssigned' in param_identity.get('type') and current_managed_type == 'SystemAssigned':
+            pass
+        # If type set to SystemAssigned, and Resource has current identity, remove UserAssigned identity
+        elif param_identity.get('type') == 'SystemAssigned':
+            changed = True
+        # If type in module args contains 'UserAssigned'
+        elif 'UserAssigned' in param_identity.get('type'):
+            if param_managed_identity not in current_managed_identity.keys():
+                user_identity = { param_managed_identity: {} }
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_aks.py]: whitespace after '{,}'
```suggestion
                user_identity = {param_managed_identity: {}}
```

File: plugins/modules/azure_rm_aks.py [hunk 5]
```diff
@@ -870,6 +908,14 @@ def __init__(self):
                 type='dict',
                 options=service_principal_spec
             ),
+            identity=dict(
+                type='dict',
+                options=managed_identity_spec,
+                required_if = [
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_aks.py]: unexpected spaces around keyword / parameter equals
```suggestion
                required_if=[
```

**Review by Fred-sun — COMMENTED** [review]
@p3ck small change request!

**Comment by p3ck** [comment]
Thanks for the feedback.  I will be sure to run sanity tests from now on.

**Comment by Fred-sun** [comment]
@p3ck No further questions, I will push for merger. Thank you!

**Review by xuzhang3 — APPROVED** [review]

**Pull request merged**